client/mysqldump.cc

```diff
@@ -174,17 +175,75 @@ static uint opt_use_gtid;
 static uint my_end_arg;
 static char * opt_mysql_unix_port=0;
 static int   first_error=0;
-/*
-  multi_source is 0 if old server or 2 if server that support multi source 
-  This is chosen this was as multi_source has 2 extra columns first in
-  SHOW ALL SLAVES STATUS.
-*/
-static uint multi_source= 0;
+static bool have_is_slave_status;
 static DYNAMIC_STRING extended_row;
 static DYNAMIC_STRING dynamic_where;
 static MYSQL_RES *get_table_name_result= NULL;
 static MEM_ROOT glob_root;
-static MYSQL_RES *routine_res, *routine_list_res, *slave_status_res= NULL;
+static MYSQL_RES *routine_res, *routine_list_res= NULL;
+
+static int mysql_query_with_error_report(MYSQL *, MYSQL_RES **, const char *);
+template<typename F> struct Slave_status_list
```

Comment:
This is a very general interface which applies to pretty much all querying. I'm not necessarily against the idea, but for this use case, I think it doesn't make sense to define here (as once `SSS` support is removed, it would be much less effective). Perhaps it could be discussed with the connector team to add support for something like this.